nautobot/core/ui/object_detail.py

```diff
@@ -484,17 +499,34 @@ def get_extra_context(self, context):
         for listing and adding objects. It also handles field inclusion/exclusion and
         displays the appropriate table title if provided.
         """
-        body_content_table = context[self.table_key]
+        body_content_table_class = self.table_class
+        body_content_table_model = body_content_table_class.Meta.model
         request = context["request"]
+        instance = get_obj_from_context(context)
+        body_content_table_queryset = body_content_table_model
+        if self.table_attribute:
+            object_manager = getattr(instance, self.table_attribute)
+        if self.table_filter:
+            object_manager = body_content_table_model.objects.filter(**{self.table_filter: instance})
+        body_content_table_queryset = (
+            object_manager.restrict(request.user, "view")
+            .order_by(*self.order_by_fields or [])
+        )
```

Comment:
I think the fallback here might result in removing the default ordering if no order_by_fields are set? Not sure, but maybe safer:

```suggestion
        body_content_table_queryset = object_manager.restrict(request.user, "view")
        if self.order_by_fields:
            body_content_table_queryset = body_content_table_queryset.order_by(*self.order_by_fields)
```